lib/core/src/text/mod.rs

```diff
@@ -0,0 +1,267 @@
+        );
+
+        unsafe { // Note [unsafe buffer_data]
+            let float_32_array = Float32Array::view(&vertices);
+            gl_context.buffer_data_with_array_buffer_view(
+                WebGlRenderingContext::ARRAY_BUFFER,
+                &float_32_array,
+                WebGlRenderingContext::STATIC_DRAW,
+            );
+        }
+
+        buffer
+    }
+
+    /* Note [unsafe buffer_data]
+     *
+     * The Float32Array::view is safe as long there are no allocations done
+     * until it is destroyed. This way of creating buffers were taken from
+     * wasm-bindgen examples
+     * (https://rustwasm.github.io/wasm-bindgen/examples/webgl.html)
+     */
+
+    fn create_vertex_buf(&self, gl_context : &Context) -> WebGlBuffer {
+        let y_max = self.y  + self.size;
+        let x_step = self.size;
+        let vertices = (0..self.text.len()).map(|i| {
+            let ix      = self.x + (i as f32) * x_step;
+            let ix_max  = ix + x_step;
+            vec![ix, self.y, ix, y_max, ix_max, self.y, ix_max,
+                 self.y, ix, y_max, ix_max, y_max] // Note [The ugly code]
+        }).flatten().collect::<Box<[f32]>>();
+
+        Self::create_buffer(gl_context, vertices.as_ref())
+    }
+
+    /* Note [The ugly code]
+     *
+     * I have already refactored this function on branch
+     * wip/ao/letter-alignment. Please be patient
+     */
+
+    fn create_tex_coord_buf(&mut self, gl_context : &Context) -> WebGlBuffer {
+        let font = &mut self.font;
+        for ch in self.text.chars() {
+            font.get_or_create_char_info(ch);
+        }
+        let vertices = self.text.chars().map(|c| {
+            let msdf_rows = font.msdf_texture.rows() as f32;
+            let info = font.get_or_create_char_info(c);
+            let y_min = info.msdf_texture_rows.start as f32 / msdf_rows;
+            let y_max = info.msdf_texture_rows.end as f32 / msdf_rows;
+            vec![0.0, y_min, 0.0, y_max, 1.0, y_min,
+                 1.0, y_min, 0.0, y_max, 1.0, y_max]
+        }).flatten().collect::<Box<[f32]>>();
+
+        Self::create_buffer(gl_context, vertices.as_ref())
+    }
+
+    fn create_msdf_texture(&self, gl_context : &Context, gl_program : &Program)
+        -> WebGlTexture {
+
+        let msdf_texture = gl_context.create_texture().unwrap();
+        gl_context.bind_texture(Context::TEXTURE_2D, Some(&msdf_texture));
+
+        gl_context.tex_parameteri(
+            Context::TEXTURE_2D,
+            Context::TEXTURE_WRAP_S,
+            Context::CLAMP_TO_EDGE as i32
+        );
+        gl_context.tex_parameteri(
+            Context::TEXTURE_2D,
+            Context::TEXTURE_WRAP_T,
+            Context::CLAMP_TO_EDGE as i32
+        );
+        gl_context.tex_parameteri(
+            Context::TEXTURE_2D,
+            Context::TEXTURE_MIN_FILTER,
+            Context::LINEAR as i32
+        );
+
+        gl_context.tex_image_2d_with_i32_and_i32_and_i32_and_format_and_type_and_opt_u8_array(
```

Comment:
I just wanted to write here >80 chars, but I stopped in mid-typing 😂

Reply:
Nice name btw

Reply:
76 characters. I _could_ try to fit in line, but I though that it would look worse

Reply:
@wdanilo and you criticized me for `absolutePosition` :P